Settings UI

weisJ opened the pull request:
Implements the settings ui. #91 

The settings can be found under `File->Settings`. Currently only the theme settings are relevant, but I have added
another dummy page with multiple different properties to showcase the capabilities of the system.

~~I'll add some pictures to the PR later.~~
As promised here a gif of the settings panel in action. This is is a test dialog and doesn't include "Cancel" and "OK" buttons.
![settings_ui](https://user-images.githubusercontent.com/31143295/101245350-870f6380-370c-11eb-9b5d-c8f9b3ba5aad.gif)

weisJ committed:
Add method to create more specific views using the ViewFactory

weisJ committed:
Male property bound condition more efficient and reliable.

weisJ committed:
Adjust settings api to be more suitable for automatically created UIs.
- Extract base Presentation class for future reuse.
- Add actions specific presentation information to ActionPresentation.
- Add convenience methods to check whether a settings container/group is empty.
- Add displayName property to SettingsGroup
- Make use of kotlin dsl markers

weisJ committed:
Port layout dsl from IntelliJ to be independent of the IntelliJ platform.
This port has a reduced feature set (i.e. cell mode isn't available) compared to the original version, but
suffices for our purposes.

weisJ committed:
Implement SettingsPanel which auto generates a view binding for given SettingsContainers.
For the ThemeSettings the panel provided by darklaf is incorporated into the ui using a separate ViewProvider.

Add option to change the themes used in os settings mode.

weisJ committed:
Fix actions of menu items not being performed.

weisJ committed:
Add SettingsDialog.
Make settings menu item display the SettingsDialog

weisJ committed:
Update sonarlint to version 1.1.4
Fix new sonarlint problems.

marko-radosavljevic reviewed (approved):
Everything is coming together! ^^

Great work. :rocket:

weisJ committed:
Update workflows:
- Remove redundant `checkout HEAD^2` command.
- Update sonarlint plugin version
- Increase metaspace for sonarlint task
